middleware/logging.go

```diff
@@ -1,4 +1,4 @@
-package logging
+package middleware
```

Comment:
Our library? This is currently not structured as a library for anyone to import. We could consider moving all API related stuff to the api folder and then have subfolders such as middleware and controllers within.

Reply:
OK, our package. As everyone can import our package (because go) we still need to consider what should go into "internal" directories, to follow go concepts of private code from the start (and understand what that internal stuff is supposed to be).

We need to establish if our top root directory represents the api or contains the api.
Before your notice that you consider it to be the api (and thus api directory making no sense) I used to think that it contains the api. But maybe your understanding is better.

Then we can figure out where the middleware must go.

Reply:
I was a bit unsure about moving stuff to the root directory, but it seemed to be what most projects did. My though was then to make the API the top level directory and use internal for stuff like computations and data fetching.

If anyone was to import our package I think it would be the stuff in internal/core they would be interested in. If we are to do a top level split I think it would make more sense to split the API code (everything that deals with the web interface) and the code in internal/core which deals with computations and data fetching.